pkg/unikontainers/unikernels/linux.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2023-2025, Nubificus LTD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unikernels
+
+import (
+	"fmt"
+	"strings"
+)
+
+const LinuxUnikernel string = "linux"
+
+type Linux struct {
+	App        string
+	Command    string
+	Env        []string
+	Net        LinuxNet
+	RootFsType string
+}
+
+type LinuxNet struct {
+	Address string
+	Gateway string
+	Mask    string
+}
+
+func (l *Linux) CommandString() (string, error) {
+	rdinit := ""
+	bootParams := "panic=-1 console=ttyS0"
+	if l.RootFsType == "block" {
+		rootParams := "root=/dev/vda rw"
+		bootParams += " " + rootParams
+	} else if l.RootFsType == "initrd" {
+		rootParams := "root=/dev/ram0 rw"
+		rdinit = "rd"
+		bootParams += " " + rootParams
+	}
+	if l.Net.Address != "" {
+		netParams := fmt.Sprintf("ip=%s::%s:%s:urunc:eth0:off",
+			l.Net.Address,
+			l.Net.Gateway,
+			l.Net.Mask)
+		bootParams += " " + netParams
+	}
+	for _, eVar := range l.Env {
+		bootParams += " " + eVar
+	}
+	if l.App != "" {
+		initParams := rdinit + "init=" + l.App + " -- " + l.Command
+		bootParams += " " + initParams
+	}
+
+	return bootParams, nil
+}
+
+func (l *Linux) SupportsBlock() bool {
+	return true
+}
+
+func (l *Linux) SupportsFS(_ string) bool {
+	return true
+}
+
+func (l *Linux) MonitorNetCli(_ string) string {
+	return ""
+}
+
+// Mewz does not seem to support virtio block or anu other kind of block/fs.
+func (l *Linux) MonitorBlockCli(monitor string) string {
+	switch monitor {
+	case "qemu":
+		bcli := " -device virtio-blk-pci,id=blk0,drive=hd0"
+		bcli += " -drive format=raw,if=none,id=hd0,file="
+		return bcli
+	default:
+		return ""
+	}
+}
+
+// Mewz does not require any monitor specific cli option
```

Comment:
Same as above, some comments are referencing Mewz, is this intentional?

Reply:
Thank you for pointing it out!